operator/config/mapkubeapis/embed_map.go

```diff
@@ -0,0 +1,45 @@
+package mapkubeapis
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+const tempFilePattern = "mapkubeapis-map-*.yaml"
+
+var (
+	//go:embed Map.yaml
+	embeddedMapFile []byte
+
+	tempMapFile string
+
+	once sync.Once
+)
+
+// GetOrCreateTempMapFile creates a temporary map file from the embedded FS if not exists. Returns the path to a created or already existing temporary file.
+func GetOrCreateTempMapFile() string {
```

Comment:
nit: for a caller from another library this exposes implementation details that a `TempFile` is used and that it is either `Create` or `Get`.
Imho it is sufficient to `GetMapFilePath()` because the caller does not need to know whether the file is created or not. 
Inside the package to itself the information is valuable, i.e. in line 33 as it gives the reader more insight into the implementation.

```suggestion
func GetFilePath() string {
```

Reply:
Need to call out that I appreciated the naming, better being more explicit than to less 💯

Reply:
Changed to `GetMapFilePath`